client/extensions/woocommerce/state/ui/products/edits-reducer.js

```diff
@@ -29,9 +29,14 @@ function editProductAction( edits, action ) {
 
 	const prevEdits = edits || {};
 	const bucket = product && isNumber( product.id ) && 'updates' || 'creates';
-	const _array = editProduct( prevEdits[ bucket ], product, data );
+	const _product = product || { id: { index: ( prevEdits[ bucket ] || [] ).length } };
```

Comment:
The internals of the reducers are starting to be a bit opaque - please consider well-named helper functions before expanding further.